code/modules/power/singularity/field_generator.dm

```diff
@@ -47,51 +53,35 @@ field_generator power level display
 	if(level)
 		overlays += "+p[level]"
 
-	return
-
 
 /obj/machinery/field/generator/New()
 	..()
 	fields = list()
 	connected_gens = list()
-	return
 
 
 /obj/machinery/field/generator/process()
-	if(Varedit_start == 1)
-		if(active == 0)
-			active = 1
-			state = 2
-			power = field_generator_max_power
-			anchored = 1
-			warming_up = 3
-			start_fields()
-			update_icon()
-		Varedit_start = 0
-
-	if(src.active == 2)
+	if(active == ONLINE)
 		calc_power()
 		update_icon()

```

Comment:
this was the thing about field gens that needed fixing the most.
this should be an on-demand call to ```update_icon()```, not constantly polled by ```process()```